est.nf

```diff
@@ -4,23 +4,37 @@ process get_sequence_ids {
         path 'import_stats.json', emit: 'import_stats'
         path 'sequence_metadata.tab', emit: 'sequence_metadata'
         path 'sunburst_ids.tab', emit: 'sunburst_ids'
+        path 'blast_hits.tab', emit: 'blast_hits'
     stub:
     """
     cp $existing_fasta_file allsequences.fa
     """
     script:
-    common_args = "--efi-config ${params.efi_config} --efi-db ${params.efi_db} --mode ${params.import_mode}"
-    if (params.import_mode == "family")
+    common_args = "--efi-config ${params.efi_config} --efi-db ${params.efi_db} --mode ${params.import_mode} --sequence-version ${params.family_id_format}"
+    if (params.import_mode == "family") {
         """
-        perl $projectDir/src/est/import/get_sequence_ids.pl $common_args --family ${params.families} --sequence-version ${params.family_id_format}
+        perl $projectDir/src/est/import/get_sequence_ids.pl $common_args --family ${params.families}
         """
-    else if (params.import_mode == "accessions") {
+    } else if (params.import_mode == "accessions") {
         """
-        perl $projectDir/src/est/import/get_sequence_ids.pl $common_args --accessions ${params.accessions_file}
+        perl $projectDir/src/est/import/get_sequence_ids.pl $common_args --user-accession-file ${params.accessions_file}
         """
-    }
-    else
+    } else if (params.import_mode == "blast") {
+        // blast_hits.tab is provided as an output to the user
+        """
+        blastall -p blastp -i ${params.blast_query_file} -d ${params.fasta_db} -m 8 -e ${params.blast_evalue} -b ${params.num_blast_matches} -o init.blast
+        SZ=\$(wc -c "init.blast" | cut -d' ' -f1)
+        if [[ \$SZ != 0 ]]; then
```

Comment:
My understanding is that `-s` is a bash extension specific to Linux.  `stat` is likewise Linux-specific.  Those may or may not be present in OS X, for example.  This may be a non-issue.

Reply:
The process script is interpreted by Nextflow as a Bash script by default
https://www.nextflow.io/docs/latest/process.html#scripts-a-la-carte

The docker image is meant to be used if someone is running this locally so their underlying platform should not matter.